src/test/javascript/spec/component/exam/manage/exams/exam-detail.component.spec.ts

```diff
@@ -0,0 +1,120 @@
+import * as chai from 'chai';
+import * as sinonChai from 'sinon-chai';
+import * as sinon from 'sinon';
+import { ComponentFixture, fakeAsync, TestBed } from '@angular/core/testing';
+import { TranslatePipe } from '@ngx-translate/core';
+import { MockComponent, MockDirective, MockPipe, MockProvider } from 'ng-mocks';
+import { AlertComponent } from 'app/shared/alert/alert.component';
+import { ActivatedRoute, Data } from '@angular/router';
+import { JhiTranslateDirective } from 'ng-jhipster';
+import { FaIconComponent } from '@fortawesome/angular-fontawesome';
+import { RouterTestingModule } from '@angular/router/testing';
+import { By } from '@angular/platform-browser';
+import { Location } from '@angular/common';
+import { ExamDetailComponent } from 'app/exam/manage/exams/exam-detail.component';
+import { AlertErrorComponent } from 'app/shared/alert/alert-error.component';
+import { Exam } from 'app/entities/exam.model';
+import { AccountService } from 'app/core/auth/account.service';
+import { ArtemisMarkdownService } from 'app/shared/markdown.service';
+import { ArtemisDatePipe } from 'app/shared/pipes/artemis-date.pipe';
+import { Course } from 'app/entities/course.model';
+import { Component } from '@angular/core';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+@Component({
+    template: '',
+})
+class DummyComponent {}
+
+describe('ExamDetailComponent', () => {
+    let examDetailComponentFixture: ComponentFixture<ExamDetailComponent>;
+    let examDetailComponent: ExamDetailComponent;
+
+    const exampleHTML = '<h1>Sample Markdown</h1>';
+    const exam = new Exam();
+    exam.id = 1;
+    exam.course = new Course();
+    exam.course.id = 1;
+    exam.title = 'Example Exam';
```

Comment:
Is it better to instantiate objects with `new`? Why not just:

```
const course: Course = {id = 1};
const exam: Exam = {id: 1, title: 'Example Exam', course};
```

Reply:
Fwiw, we write it like the following in a lot of places:
```
const examCourse = { id: 1 } as Course;
const exam = { id: 1, title: 'Example Exam', course: examCourse } as Exam;
```

Reply:
I mean this seems to be a matter of opinion :D . I like the typescript class constructor method because sometimes stuff like types are set by the constructor. you will miss that if you use type declaration